pkgs/firehose/lib/src/pubspec.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart' as yaml;
+
+class Pubspec {
+  final Directory directory;
+  late final Map _yaml;
+
+  Pubspec(this.directory) {
+    var file = File(path.join(directory.path, 'pubspec.yaml'));
+    _yaml = yaml.loadYaml(file.readAsStringSync()) as Map;
+  }
+
+  /// Return the package name.
+  String get name => _yaml['name'] as String;
+
+  /// Return the package version.
```

Comment:
[nit] no need to describe a getter as "Return"

/// The package version.
///
/// Null if no version is specified.

Similar throughout.